six/modules/python/six/source/six.i

```diff
@@ -100,6 +102,40 @@ def setValue(self, *args):
 %}
 
 
+%extend nitf::DateTime
+{
+  public:
+    PyObject* toPythonDateTime()
+    {
+        PyObject* nitfDateTime = NULL;
+        int year = self->getYear();
+        int month = self->getMonth();
+        int day = self->getDayOfMonth();
+        int hour = self->getHour();
+        int minute = self->getMinute();
+        int second = static_cast<int>(self->getSecond());
+        int microsecond = static_cast<int>((self->getSecond() - second) * 1e6);
+        nitfDateTime = PyDateTime_FromDateAndTime(year, month, day,
+                hour, minute, second, microsecond);
+        return nitfDateTime;
+    }
+
+    %pythoncode
+    %{
+        @staticmethod
+        def fromPythonDateTime(pyDatetime):
+            second = pyDatetime.second + (pyDatetime.microsecond / 1e6);
+            sixDatetime = DateTime()
```

Comment:
You can actually use the `DateTime` constructor that lets you send in all these things in upfront.  This will be (very slightly) more efficient since the default constructor gets the current date/time and there's no need to do that.